src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1568,6 +1577,63 @@ func (k *kataAgent) handleDeviceBlockVolume(c *Container, m Mount, device api.De
 	default:
 		return nil, fmt.Errorf("Unknown block device driver: %s", c.sandbox.config.HypervisorConfig.BlockDeviceDriver)
 	}
+	return vol, nil
+}
+
+func handleImageGuestPullBlockVolume(c *Container, virtualVolumeInfo *types.KataVirtualVolume, vol *grpc.Storage) (*grpc.Storage, error) {
+	container_annotations := c.GetAnnotations()
+	container_type := container_annotations["io.kubernetes.cri.container-type"]
+	if virtualVolumeInfo.Source == "" {
+		var image_ref string
+		if container_type == "sandbox" {
+			image_ref = "pause"
+		} else {
+			image_ref = container_annotations["io.kubernetes.cri.image-name"]
```

Comment:
The annotations used here are from containerd. cri-o will use different ones.
I understand that this is just copying the existing code from CCv0, but can we add the cri-o annotation support here, or should it be a separate PR?

Reply:
Thanks, @littlejawa. I would add the cri-o annotation support here.